lib/Base/Tensor.cpp

```diff
@@ -379,6 +380,60 @@ std::string Tensor::toString(unsigned maxNumElem) const {
   return os.str();
 }
 
+void Tensor::dumpToRawBinaryFile(const char *filename) {
+  std::ofstream fs;
+  fs.open(filename, std::ios::out | std::ios::binary);
+  DCHECK(fs.is_open()) << "Failed to open file: " << filename;
+  auto handle = getHandle<>();
+  for (dim_t rdi = 0; rdi < actualSize(); rdi++) {
+    fs.write((char *)&handle.raw(rdi), sizeof(handle.raw(rdi)));
+  }
+  fs.close();
+}
+
+void Tensor::dumpToRawTextFile(const char *filename) {
```

Comment:
Dumping into a text file should probably use hex to save the precise values. Dumping floating point values using a decimal format is not reliable and may loose precision.

Reply:
But textual representation serves the purpose of readability. That is why I added the two flavors: binary to promote bit exactness, text to promote readability. I enhanced the method description to point that.